### ISSUE TITLE
Add Kotlin as a optional dependency of core module.

### ISSUE THREAD

**Issue by ikhoon** [line/armeria#4229]
If Kotlin dependency is added to an optional dependency, we can provide Kotlin-specific extension methods and DSL for each module.
For example:
```kt
// WebClientExtensions.kt
inline fun <reified T> WebClientRequestPreparation.asJson():
        FutureTransformingRequestPreparation<ResponseEntity<T>> {
    return asJson(object : TypeReference<T>() {})
}

val response: ResponseEntity<Item> = 
  client.prepare()
        .get("/example")
        .asJson<Item>()  // Simply specify the target type parameter.
        .execute()
```

**Comment by Jimexist** [comment]
by the way i added [kotlin support][1] in thrift some time ago, could benefit from that once libthrift and compiler 0.17.0 is released in Q3 2022

[1]: https://github.com/apache/thrift/blob/master/compiler/cpp/src/thrift/generate/t_kotlin_generator.cc

**Comment by ikhoon** [comment]
It looks to support Kotlin Coroutines. Awesome! ❤️
https://github.com/apache/thrift/blob/master/compiler/cpp/src/thrift/generate/t_kotlin_generator.cc#L1393

**Comment by Jimexist** [comment]
> It looks to support Kotlin Coroutines. Awesome! ❤️ https://github.com/apache/thrift/blob/master/compiler/cpp/src/thrift/generate/t_kotlin_generator.cc#L1393

yes i intended to only support suspend functions, but then delegate to jdk8 bridge for suspend function <=> async client operations, so that the generated code can still work with the existing Java libthrift library _alone_, i.e. there's no libthrift-kotlin library.

i can see that in armeria the thrift client is actually not based on the default implementation but rather via [dynamic proxy and delegate][1], so ideally this can be done similarly for kotlin

[1]: https://github.com/line/armeria/blob/master/thrift0.13/src/main/java/com/linecorp/armeria/internal/client/thrift/THttpClientDelegate.java#L72
